app/components/Quest/Carousel.tsx

```diff
@@ -0,0 +1,179 @@
+                    <div className=" w-60 md:w-full max-w-2xl grid grid-cols-1 md:grid-cols-3 place-content-center justify-items-center  gap-x-8">
+                    {badges.map((image, index) => (
+                        <div key={`image_${index}`} className="relative px-2 flex justify-center items-center outline-0 my-2">
+                            <img alt={`Carbonable Badge ${index}`} onMouseOver={() => handleClick(index)} src={`/assets/images/quest/${image.name}`} className={index === activeSlide ? "rounded-lg brightness-50  w-full h-40 z-0 " : "rounded-lg  w-full h-40 z-0"}   />
+                            { (image.mintable && index === activeSlide )   &&
+
+                            <div className="absolute  h-full bg-green z-20 uppercase font-inter font-bold  text-black  w-11/12 py-2 px-2 top-0  text-[8px] md:text-xs  lg:px-3 "> 
+                                <div className="grid grid-flow-row  h-full items-stretch">
+                                    <p className="font-trash font-bold text-3xl self-start">green <br /></p> 
+                                    <p className="font-americana font-thin text-2xl self-start">pioneer</p>
+                                    <BadgeMint className=" place-self-center self-end w-28" onClick={async () => {
+                                        setMenu(<LoadingScreen />)
+
+                                        if (!badgeContract) return;
+                                        if (!account) return setMenu(<WalletMenu action={() => setMenu(null)} />);
+
+                                        // Check if the user has already minted the badge
+                                        const balance = parseInt((await badgeContract.functions.balanceOf(address, [image.token_id, 0])).balance.low);
+                                        if (balance > 0) return setMenu(<ErrorMessage strong="You already have minted this badge." text="You can only mint each badge one time." action={() => setMenu(null)} />);
+
+                                        // Check if the user is whitelisted
+                                        const res = await fetch(`quest/sign/${account.address}/${image.token_id}`)
+                                        const signature: Signature = await res.json();
+                                        if (!signature.low || !signature.high) return setMenu(<ErrorMessage strong="You are currently not in the whitelist." text="Complete the quests on Crew3, then wait a bit and you'll be added." action={() => setMenu(null)} />);
+
+                                        // Mint the badge
+                                        setMenu(<SuccessMessage strong="Success." text="Please approve the transaction" action={() => setMenu(null)} />)
+                                        setSignature(signature);
+                                        setBadgeType(image.token_id);
```

Comment:
For better clarity, it's better to put this bloc in a dedicated function. 

Something like:
onClick={handleClick()}

const handle click = async () => {
// The code here
}

Reply:
<img width="1006" alt="CleanShot 2022-12-27 at 09 46 36@2x" src="https://user-images.githubusercontent.com/25151724/209639251-703492e3-a587-4277-a2f4-4081b0ca8f27.png">
tried but did'nt succeed, it may be the opportunity of a future improvement patch

Reply:
@Marchand-Nicolas by any chance if you can have a look ! for a new PR

Reply:
I can do it tomorrow